git/gitignore.go

```diff
@@ -0,0 +1,107 @@
+package git
+
+import (
+	"errors"
+	"log/slog"
+	"os"
+	"strings"
+
+	"dagger.io/dagger"
+)
+
+type Paths struct {
+	Files       []string
+	Directories []string
+	Globs       []string
+}
+
+type Include struct {
+	Files       map[string]dagger.File
+	Directories map[string]dagger.Directory
+}
+type Gitignore struct {
+	Include Include
+	Exclude Paths
+}
+
+func IsDirectory(path string) (bool, bool, error) {
+	fileInfo, err := os.Stat(path)
+	if errors.Is(err, os.ErrNotExist) {
+		return false, false, nil
+	}
+	if err != nil {
+		return false, false, err
+	}
+
+	return fileInfo.IsDir(), true, err
+}
+
+func DeterminePathType(path string, basePath string, paths *Paths) {
```

Comment:
Pretty sure most or if not all of this function could be replaced with some calls to functions in the `filepath` package ;)